### ISSUE TITLE
Automation Account creation: AA name and RG name swapped, introduced with AzureRM 2.88.

### ISSUE THREAD

**Issue by mivica** [hashicorp/terraform-provider-azurerm#14465]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

- Terraform v1.0.11 on windows_amd64
- Installed hashicorp/azurerm v2.88.0

### Affected Resource(s)

- azurerm_automation_account

### Terraform Configuration Files

```hcl

resource "azurerm_automation_account" "test_automation_account" {
  name                = "aa-test"
  resource_group_name = "rg-test"
  location            = "North Europe"
  sku_name            = "Basic"
}

```

### Debug Output

│ Error: creating/updating Automation Account: (Name "rg-test" / Resource Group "aa-test"): automation.AccountClient#CreateOrUpdate: Failure responding to request: StatusCode=404 -- Original Error: autorest/azure: Service returned an error. Status=404 Code="ResourceGroupNotFound" Message="Resource group 'aa-test' could not be found."
│
│   with azurerm_automation_account.test_automation_account,
│   on main.tf line 1, in resource "azurerm_automation_account" "test_automation_account":
│    1: resource "azurerm_automation_account" "test_automation_account" {
│

### Expected Behaviour

Resource azurerm_automation_account is created in the desired resource group.

### Actual Behaviour

Creation of azurerm_automation_account resource fails. It looks like if the resource provider is mixing AA name with RG name, looking for the wrong RG.

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

This is all working fine with azurerm v2.87.0. The issue appears to be introduced with azurerm v2.88.0.

### References

* #0000

**Comment by l-lafin** [comment]
The same issue is happening when retrieving an automation account through `data`.

**Comment by misspellted** [comment]
Seems to be a result of https://github.com/hashicorp/terraform-provider-azurerm/commit/4394fb138bded3f8ab4289aa6e24d808af0c9af2...

Is there an intent to reorder the parameters, and the automation account just happened to be the first one attempted?

**Comment by tombuildsstuff** [comment]
Fixed via #14464
